### ISSUE TITLE
mailgun-ruby-1.1.3.gem does not match published SHA

### ISSUE THREAD

**Issue by mwpastore** [mailgun/mailgun-ruby#76]
The mailgun-ruby-1.1.3.gem on rubygems.org is published with a SHA that does not match the SHA of the file. This could potentially indicate that the contents of the gem have been manipulated in some way. 

```
Bundler cannot continue installing mailgun-ruby (1.1.3).
The checksum for the downloaded `mailgun-ruby-1.1.3.gem` does not match the checksum given by the server. This means the contents of
the downloaded gem is different from what was uploaded to the server, and could be a potential security issue.

To resolve this issue:
1. delete the downloaded gem located at:
`/Users/mwp/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/mailgun-ruby-1.1.3/mailgun-ruby-1.1.3.gem`
2. run `bundle install`

If you wish to continue installing the downloaded gem, and are certain it does not pose a security issue despite the mismatching
checksum, do the following:
1. run `bundle config disable.checksum_validation true` to turn off checksum verification
2. run `bundle install`

(More info: The expected SHA256 checksum was "2542b467788ab91479a84617f9aeeb02c1446a52229d80d0a857bed20839d80c", but the checksum
for the downloaded gem was "da81bdb633549e3035252fe1d89f967396bff93d1a7b194c7ba424cbb906c2ad".)
```

**Comment by pirogoeth** [comment]
@mwpastore Thank you for the report! I'll look in to this asap.

**Comment by pixeltrix** [comment]
@pirogoeth the reason for the mismatch is that your Travis build tries to push the gem for every ruby version in your build matrix. 99 times out of 100 this doesn't cause a problem but looks like there might be a race condition in rubygems.org that two pushes at the same time will fail and the wrong build file gets put into the S3 bucket.

I'd suggest only pushing from one build to be on the safe side in future.

**Comment by pirogoeth** [comment]
@pixeltrix I figured that's what it was. I'll fix it asap. Thanks!

**Comment by mwpastore** [comment]
Thank you both!

**Comment by dwradcliffe** [comment]
FYI I've replaced the .gem from the first gem push so the SHA for version 1.1.3 will match now. We're also going to look to fix the race condition.

**Comment by pirogoeth** [comment]
@dwradcliffe Thanks!

**Comment by pirogoeth** [comment]
This is fixed in the deploy changes from #78
